atomsci/ddm/pipeline/splitting.py

```diff
@@ -710,7 +710,8 @@ def compact_dataset(self):
             # w = [[0, 1, 0], --> w = [[1, 1, 0]]
             #      [1, 0, 0],
             #      [1, 0, 0]]
-            w_agg_func = lambda x: np.clip(np.sum(x, axis=0), a_min=0, a_max=1)
+            def w_agg_func(x):
+                np.clip(np.sum(x, axis=0), a_min=0, a_max=1)
```

Comment:
This is something we can ask the core AMPL team on their preferences. If they want to disable this rule, then we can add that setting to `ruff.toml`